File: Loenn/entities/crystalbomb.lua
local drawableSprite = require("structs.drawable_sprite")
local utils = require("utils")
local crystalbomb = {}

crystalbomb.name = "cavern/crystalbomb"
crystalbomb.depth = -9999
crystalbomb.texture = "objects/cavern/crystalbomb/idle00"
crystalbomb.justification = {0.5, 0.5}
crystalbomb.placements = 
{
    name = "CrystalBomb",
    data = 
    {
        ["respawnTime"] = 2.0,
        ["explodeTime"] = 1.0,
        ["explodeOnSpawn"] = false,
        ["respawnOnExplode"] = true,
        ["breakDashBlocks"] = false
    }
}

function crystalbomb.rectangle(room, entity)
    local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 8, y - 3, 17, 13)
end

return crystalbomb

File: Ahorn/entities/crystalBomb.jl
module CavernCrystalBomb

using ..Ahorn, Maple

@mapdef Entity "cavern/crystalbomb" CrystalBomb(x::Integer, y::Integer, respawnTime::Number=2.0, explodeTime::Number=1.0, explodeOnSpawn::Bool=false, respawnOnExplode::Bool=true, breakDashBlocks::Bool=false, legacyMode::Bool=false)

const placements = Ahorn.PlacementDict(
    "Crystal Bomb (Cavern)" => Ahorn.EntityPlacement(
        CrystalBomb,
        "point"
    )
)

sprite = "objects/cavern/crystalbomb/idle00.png"

function Ahorn.selection(entity::CrystalBomb)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 10, y - 14, 19, 15)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CrystalBomb) = Ahorn.drawSprite(ctx, sprite, 0, -10)

end